File: 3754-concatenate-non-zero-digits-and-multiply-by-sum-i/3754-concatenate-non-zero-digits-and-multiply-by-sum-i.py
class Solution:
    def sumAndMultiply(self, n: int) -> int:
        s = str(n)
        res = []
        sum = 0
        for i in range(len(s)):
            if s[i] == '0':
                continue
            res.append(s[i])
            sum += int(s[i])
        
        if len(res) == 0: return 0
        
        return int(''.join(res)) * sum

# math
# time: O(n)
# space: O(n)

File: 3754-concatenate-non-zero-digits-and-multiply-by-sum-i/3754-concatenate-non-zero-digits-and-multiply-by-sum-i.go
func sumAndMultiply(n int) int64 {
    strNum := strconv.Itoa(n)
    sum := int64(0)
    res := []byte{}
    for i := 0; i < len(strNum); i++ {
        if strNum[i] == '0' {
            continue
        }
        sum += int64(strNum[i] - '0')
        res = append(res, strNum[i])
    }

    intRes, _ := strconv.Atoi(string(res))
    return int64(intRes) * sum
}

// math
// time: O(n)
// space: O(n)